proposals/NNNN-widget-support.md

```diff
@@ -0,0 +1,248 @@
+
+Details about the currently selected screen template are part of the display capabilities struct:
+- text fields, the name, width etc.
+- image fields e.g. graphic, or soft button image resolutions
+
+The template design comes from the head unit. However predefined template names are defined in `PredefinedLayout` enum. The head unit should provide template designs for the predefined template names. Still it can also provide additional custom templates with self defined names. 
+
+All this can be reused for widget screens. When creating a new widget screen the available widget templates are provided in `CreateScreenResponse.displayCapabilities.templatesAvailable`.
+
+The details about the currently selected widget template are provided using the same display capabilities struct. It is expected that the HMI will return with fewer text fields (only main field 1 and 2), shorter text field width and smaller image resolutions. This proposed solution reuses the existing API to provide widget template details mentioned. Just as today the design for widgets also come from the head unit and the OEM.
+
+Examples for widget screens:
+
+| template name | Example layout |
+|-----------------------|----------------------------------------------------------------------------------------------|
+| text with graphic  | ![template](../assets/proposals/NNNN-widgets/template-text-with-graphic.png) |
+| graphic with text  | ![template](../assets/proposals/NNNN-widgets/template-graphic-with-text.png) |
+| tiles with graphic | ![template](../assets/proposals/NNNN-widgets/template-tiles-with-graphic.png) |
+
+### Screen types supported
+
+In order to inform the app what screen types are supported the struct `DisplayCapabilities` should be extended with a single optional parameter called `screenTypesSupported`.
+
+```xml
+<struct name="DisplayCapabilities" since="2.0">
+ :
+ <param name="screenTypesSupported" type="ScreenType" array="true" minsize="1" mandatory="false" />
+</struct>
+```
+
+This parameter does not inform the app how many screens of the type are supported but it informs if a display supports widgets or a main app screen. The HMI decides on 
+- how many widgets should be displayed (example: limited to 6 or indefinite using a scrollable area) 
+- and how to order them (alphabetically, chronogically or based on recent app usage).
+
+## Potential downsides
+
+With current design displays and screens are not clearly separated. Together with multi-display support there will be a proposal to redesign display and screen capabilities. This proposal should still be valid as it focuses on mutli screen only.
+
+## Impact on existing code
+
+On SDL there are additional RPCs to be added. The HMI needs to design templates for widgets. This proposal did not focus on the mobile changes. For mobile libraries it's necessary to add the new RPCs and add support for multiple screen managers. This will also be part of a separate proposal.
```

Comment:
Change to:
On SDL there are additional RPCs **that need to be added.** The HMI needs to design templates for widgets. This proposal did not focus on the mobile changes. For mobile libraries it's necessary to add the new RPCs and add support for multiple screen managers. This will also be part of a separate proposal.